source/tomopy/misc/corr.py

```diff
@@ -267,6 +268,137 @@ def median_filter_cuda(arr, size=3, axis=0):
     return out
 
 
+def find_nonfinite_values_prj(projection, slc_idx):
+    """Allows the user to easily obtain the Z, x, y coordinates of nonfinite 
+       values for a given 2D projection in a 3D array.
+
+    Parameters
+    ----------
+    projection : 2D nd.array
+        The 2D projection for a given 3D data array.
+
+    slc_idx : int
+        The projection index inside the main 3D array.
+
+    Returns
+    -------
+    The indices for the current projection used, the x coordinates, and y
+    coordinates of all non-finite values.
+    """
+
+    # Determining where the nonfinite values are
+    boolean_of_nonfinite = ~np.isfinite(projection)
+
+    # Obtaining there x, y locations
+    idx_of_nonfinite_prj = np.nonzero(boolean_of_nonfinite)
+
+    # Creating an identical shaped array for the z position in the 3D array the prj comes from
+    idx_of_nonfinite_data = np.asarray(
+        [slc_idx] * len(idx_of_nonfinite_prj[0]))
+
+    return idx_of_nonfinite_data, idx_of_nonfinite_prj[0], idx_of_nonfinite_prj[1]
+
+
+def determine_nonfinite_kernel_idxs(x_idx, y_idx, kernel, shape_x, shape_y):
+    """Determine the proper kernel bounds for a given x, y index, and kernel size.
+
+    Parameters
+    ----------
+    x_idx : int
+        The x index for a given nonfinite value.
+
+    y_idx :  int
+        The y index for a given nonfinite value.
+
+    shape_x : int
+        The max shape of the 2D projection in the x dimension.
+
+    shape_y : int
+        The max shape of the 2D projection in the y dimension.
+
+    Returns
+    -------
+    The integer kernel bounds to surround the nonfinite value with.
+    """
+
+    # Determining x kernel
+    x_idx_lower = x_idx - kernel
+    x_idx_higher = x_idx + kernel + 1
+
+    if x_idx_lower < 0:
+        x_idx_lower = 0
+    if x_idx_higher > shape_x-1:
+        x_idx_higher = shape_x-1
+
+    # Determining y kernel
+    y_idx_lower = y_idx - kernel
+    y_idx_higher = y_idx + kernel + 1
+
+    if y_idx_lower < 0:
+        y_idx_lower = 0
+    if y_idx_higher > shape_y-1:
+        y_idx_higher = shape_y-1
+
+    # Forcing values to be integers
+    integer_values_pre = [x_idx_lower, x_idx_higher, y_idx_lower, y_idx_higher]
+    integer_values = [int(i) for i in integer_values_pre]
+
+    return integer_values
+
+
+def median_filter_nonfinite(data, kernel=1):
+    """Apply a selective median filter with size kernel to all nonfinite
+       values in the 3D data array.
```

Comment:
Please update the description to make it clearer that the filtering is 2D (not 3D) and happens slice-by-slice along the 0th(?) axis.

Reply:
This has been removed in the latest commit (040be55 - Updating median_filter_nonfinite to not require tqdm)